backend/server.py

```diff
@@ -281,6 +388,56 @@ def extract_segments_handler(file: UploadFile = File(...)):
     except Exception as e:
         logger.error("Error in /docx2text: %s", e)
         raise HTTPException(status_code=500, detail="Failed to extract segments")
+
+@app.post('/extractText')
+async def extract_text_handler(request: Request, user_info: dict = Depends(get_user_info)):
```

Comment:
This part is redundant. You can use text docx2text from frontend and save_segments to replace this whole code.